src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,84 @@
+package Duke;
+
+import Duke.Commands.Command;
+import Duke.Storage.FileFormatException;
+import Duke.Storage.Storage;
+import Duke.Storage.TaskStorage;
+import Duke.Task.Task;
+import Duke.Task.TaskList;
+import Duke.Ui.Parser;
+import Duke.Ui.Ui;
+import Duke.Ui.UserInput;
+
+import java.io.IOException;
+
+/**
+ * The Duke class encapsulates the main logic and data for running the chat-bot.
+ *
+ * @author cai
+ */
+public class Duke {
+    /** Default path for storing tasks */
+    private static final String DEFAULT_TASK_STORAGE_PATH = "data/duke.txt";
+
+    /** Boolean indicating whether the program has been stopped */
+    private boolean isStopped = false;
+
+    /** List of tasks as added by the user */
+    private final TaskList taskList;
+
+    /**
+     * Constructs an instance of the Duke chat-bot with the specified task storage path.
+     *
+     * @param todoStoragePath Path for storing tasks.
+     * @throws IOException If IOException is thrown when creating and/or loading the tasks from the path.
+     */
+    public Duke(String todoStoragePath) throws IOException {
+        Storage<Task> taskStorage = new TaskStorage(todoStoragePath);
+        try {
+            this.taskList = new TaskList(taskStorage);
```

Comment:
Should we avoid the unnecessary use of the `this` keyword? See https://se-education.org/guides/conventions/java/index.html#variables.

I noticed this in a few other places too.

```suggestion
            taskList = new TaskList(taskStorage);
```